File: cpp/FUllTimeEmployee.cpp
#include "FullTimeEmployee.h"

FullTimeEmployee::FullTimeEmployee(const std::string& name, int id, float annualSalary)
    : Employee(name, id), annualSalary(annualSalary) {}

float FullTimeEmployee::calculateSalary() const {
    return annualSalary;
}

File: src/models/FullTimeEmployee.ts
import {Employee} from "../interface/IEmployee";

class FullTimeEmployee extends Employee {
     annualSalary: number;

    constructor(name: string, id: number, annualSalary: number) {
        super(name, id);
        this.annualSalary = annualSalary;
    }

    calculateSalary(): number {
        return this.annualSalary;
    }
}

export {FullTimeEmployee};
